### PR TITLE
Add rle *compression* and utility functions

### PR DESCRIPTION
**Pull request by jorainer**
This adds some Rle *compression* function and other utility functions that will be used across the `Spectra` and `Chromatograms` packages (see also issue #15).

### CODE REVIEW

File: DESCRIPTION [hunk 1]
```diff
@@ -19,10 +19,14 @@ Author: Laurent Gatto, Johannes Rainer and Sebastian Gibb.
...
     R (>= 3.6.0)
+Imports:
+    methods,
+    S4Vectors
 Suggests:
     testthat,
     knitr,
-    roxygen2
+    roxygen2,
+    IRanges
```
[the remark this answers is not in the document]
Comment (jorainer) [reply DESCRIPTION]: We need `IRanges` for the `NumericList` - I also thought that one is defined in `S4Vectors`, but it is not.

[the remark this answers is not in the document]
Comment (sgibb) [reply DESCRIPTION]: Ok, but you only use it in a unit tests (that's why it is in `Suggests`) but do we really need that test? Wouldn't a simple `list` be enough?

[the remark this answers is not in the document]
Comment (jorainer) [reply DESCRIPTION]: Aw, sorry, thought that is `Chromatograms` - of course, if we don't need `IRanges` I will remove it here.

**Comment by jorainer** [comment]
Thanks for the fast review. In fact the code is just copy paste and changing the names from `snake_case` to `camelCase`. We could still discuss on the naming of the functions.

File: DESCRIPTION (hunk 1 above)
[the remark this answers is not in the document]
Comment (jorainer) [reply DESCRIPTION]: `IRanges` was not removed.

**Comment by jorainer** [comment]
If depending on `S4Vectors` is OK for you @sgibb , I would merge.

**Comment by sgibb** [comment]
I will resolve the merge conflicts and push to master soon.

**Pull request merged**